### DISCUSSION TITLE
Short-circuit boolean operators

### DISCUSSION THREAD

**Discussion by dgriffie79** [onyx-lang/onyx#153] [Ideas]
boolean evaluations don't short-circuit, which is surprising. Can this be added?

**Comment by brendanfh** [comment]
This is something I have been meaning to add for a while but I kept pushing it off because I thought it was going to be harder to implement. With changes I made to some of the data structures used in the code generation step I made last year, this was actually really easy to implement, so the changes are in [this branch](https://github.com/onyx-lang/onyx/tree/feature/short-circuit) and will be merged in by the next release, and in nightly sooner than that.

**Comment by dgriffie79** [comment]
wow thanks!
